lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -45,8 +45,12 @@ class WP_Fonts_Resolver {
 	 * @return array User selected font-families when exists, else empty array.
 	 */
 	public static function enqueue_user_selected_fonts() {
-		$global_styles       = wp_get_global_styles();
-		$user_selected_fonts = static::get_user_selected_fonts( $global_styles );
+		$user_selected_fonts = array();
+		$user_global_styles  = WP_Theme_JSON_Resolver_Gutenberg::get_user_data()->get_raw_data();
```

Comment:
Would this avoid the cache we introduced?

Reply:
This has the same performance impact than using `wp_get_global_styles`. [That function has not a cache of their own](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/global-styles-and-settings.php#L109): both approaches rely on the resolver cache.